gravitee-am-test/specs/gateway/mfa/mfa-general.spec.ts

```diff
@@ -0,0 +1,345 @@
+                'Content-type': 'application/x-www-form-urlencoded',
+            },
+            302);
+
+        const authResponseFinal = await followUpGet(loginPostResponse, 302);
+
+        expect(authResponseFinal.headers['location']).toBeDefined();
+        expect(authResponseFinal.headers['location']).toContain(ctx.client.redirectUris[0]);
+        expect(authResponseFinal.headers['location']).toContain('code=');
+    });
+});
+
+describe('With one enabled factor should omit MFA flow', () => {
+    it('when stepUp=false', async () => {
+        const ctx = stepUpPositiveCtx;
+
+        const authResponse = await get(ctx.clientAuthUrl, 302);
+        const loginPage = await followUpGet(authResponse, 200);
+
+        let xsrf = extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+        let action = extractDomAttr(loginPage, 'form', 'action');
+
+        const loginPostResponse = await postForm(
+            action,
+            {
+                'X-XSRF-TOKEN': xsrf,
+                username: ctx.user.username,
+                password: ctx.user.password,
+                rememberMe: 'off',
+                client_id: ctx.client.clientId,
+            },
+            {
+                Cookie: loginPage.headers['set-cookie'],
+                'Content-type': 'application/x-www-form-urlencoded',
+            },
+            302
+        );
+
+        const authResponseFinal = await followUpGet(loginPostResponse, 302);
+
+        expect(authResponseFinal.headers['location']).toBeDefined();
+        expect(authResponseFinal.headers['location']).toContain(ctx.client.redirectUris[0]);
+        expect(authResponseFinal.headers['location']).toContain('code=');
+    });
+
+    it('when stepUp is disabled', async () => {
+        const ctx = stepUpOffCtx;
+        const authResponse = await get(ctx.clientAuthUrl, 302);
+        const loginPage = await followUpGet(authResponse, 200);
+
+        let xsrf = extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+        let action = extractDomAttr(loginPage, 'form', 'action');
+
+        const loginPostResponse = await postForm(
+            action,
+            {
+                'X-XSRF-TOKEN': xsrf,
+                username: ctx.user.username,
+                password: ctx.user.password,
+                rememberMe: 'off',
+                client_id: ctx.client.clientId,
+            },
+            {
+                Cookie: loginPage.headers['set-cookie'],
+                'Content-type': 'application/x-www-form-urlencoded',
+            }, 302);
+
+        const authResponseFinal = await followUpGet(loginPostResponse, 302);
+
+        expect(authResponseFinal.headers['location']).toBeDefined();
+        expect(authResponseFinal.headers['location']).toContain(ctx.client.redirectUris[0]);
+        expect(authResponseFinal.headers['location']).toContain('code=');
+    });
+});
+
+describe('With active session, when stepUp is true, on authorization', () => {
```

Comment:
**wish**: would love to see the test where steup is true but challenge is disabled. (step 1 : prepareMfaEnd2End, step 2: authenticate the user and go driectly on the callback with a code, step 3: on active session, initiate the login flow and the challenge should be prompted. Maybe you can create another JIRA with all the scenation not covered in this PR to remind us to implement them)